code/obj/flock/structure/rift.dm

```diff
@@ -0,0 +1,62 @@
+//
+// rift thingymabob
+//
+/obj/flock_structure/rift
+	icon = 'icons/misc/featherzone.dmi'
+	icon_state = "rift"
+	anchored = 0
+	density = 0
+	name = "glowing portal thingymabob"
+	desc = "Oh god is that a fucking light grenade?!"
+	flock_id = "Entry Rift"
+	build_time = 10
+	health = 200 // stronk little thing
+	var/decal_made = 0 // for splashing stuff on throw
+	var/list/eject = list()
+	var/mainflock = null // for when a flockmind is spawning the little shits(read:drones) get assigned to it
+
+/obj/flock_structure/rift/New(var/atom/location, var/datum/flock/F=null)
+	..()
+	if(src.flock)
+		src.flock.registerUnit(src)
+
+/obj/flock_structure/rift/building_specific_info()
+	var/time_remaining = round(src.build_time - getTimeInSecondsSinceTime(src.time_started))
+	return "Approximately <span class='bold'>[time_remaining]</span> second[time_remaining == 1 ? "" : "s"] left until entry."
+
+/obj/flock_structure/rift/process()
+	var/elapsed = getTimeInSecondsSinceTime(src.time_started)
+	if(elapsed >= build_time)
+		src.visible_message("<span class='text-blue'>Multiple shapes exit out of [src]!</span>")
+//		new /mob/living/critter/flock/drone(get_turf(src), src.flock)
+		var/j = pick(3, 4) + 1
+		for(var/i=1, i<j, i++)
+			var/obj/item/flockcache/x = new(src.contents)
+			x.resources = rand(40, 50)
+			eject += x
+		for(var/i=1, i<5, i++)
+			var/obj/flock_structure/egg/e = new(src.contents)
+			eject += e
+			e.flock = mainflock
```

Comment:
Some explanation behind this would be nice.

Reply:
so im using the flockdronegibs proc to handle throwing things around randomly. here im just creating the flockdrone eggs and resource caches before adding them to the eject list. the eject list being what gets thrown about

Reply:
can you put that in a comment?